workflow/controller/operator.go

```diff
@@ -1502,6 +1546,12 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 	}
 	if err != nil {
 		node = woc.markNodeError(node.Name, err)
+		if resolvedTmpl.Synchronization != nil {
+			woc.log.Debugf("Node %s is releasing Semaphore lock", node.ID)
```

Comment:
I see this debug message line coupled with the release in a couple places. I think this debug line should really be moved into the release function.